### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by farouqdaif1**
### In this project, I have worked on the following things 👍
✔️ Added Controllers
✔️ Added Models & Schema
✔️ Added Views
✔️ Added Devise
✔️ Made authorizations
✔️ Added splash screen
✔️ Followed Gitflow 
✔️ Followed Linters
✔️ Followed Ruby best practices
✔️ Followed Ruby on Rails best practices 
✔️ Followed HTML, and CSS best practices 
✔️ Documented work in a professional way

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,91 @@
...
-application up and running.
+"MyBudget" is a budget manager web application that allows users to keep track of their budget with different categories.
 
-Things you may want to cover:
 
-* Ruby version
+## Link Demo
 
-* System dependencies
+[Cick here to get live demo]()
```
Comment (Gambit142) [inline README.md]: - [x] Kindly deploy the project using Heroku, Netlify or GitHub pages. Please ensure that the application works well and the categories and transactions page works properly. 👍

File: README.md [hunk 2]
```diff
@@ -1,24 +1,91 @@
...
 
-* Configuration
+## Video Demo
+https://www.loom.com/share/122853d2b16a442295145af81cd49a0a
```
Comment (Gambit142) [inline README.md]: - [x] Great work recording this video and highlighting the project's features. However, you forgot to highlight a piece of code that you are proud of making. Kindly record the video again and implement this requirement. 👍

File: app/views/group/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      <h6 class="card-subtitle mb-2 text-muted"><%=group.created_at%></h6>
+      <%= link_to group.name.capitalize, group_entity_index_path(group_id: group.id) ,class:"btn btn-primary" %>
+    </div>
+  </div>
+</div>
+<% end %>
+</div>
+
+  <%= link_to "Create Category",new_group_path,class:"create-category" %>  
+</div>
```
Comment (Gambit142) [inline app/views/group/index.html.erb]: - [x] For the categories page, I should be able to see the icon I attached when creating a category. The Icon is meant to be the image for each category in the categories page according to the design template. 

image

Kindly implement this feature. 👍

File: app/views/entity/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+<ul class="list-group">
+<% @entities.each do |transaction| %>
+  <li class="list-group-item d-flex justify-content-between align-items-center">
+   <%= transaction.name %>
+    <span class="badge badge-primary badge-pill">$<%= transaction.amount %></span>
+  </li>
+<% end %>
+</ul>
+  <%= link_to "Create transcation",new_group_entity_path,class:"create-category" %>  
+</div>
```
Comment (Gambit142) [inline app/views/entity/index.html.erb]: - [x] For the transactions page kindly implement the following features:

1. For a given category the list of transactions is presented ordered by the `most recent transaction`. That is from the newest transaction to the oldest transaction. 👍 
2. At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category). 👍 

Kindly implement these features.

File: app/views/group/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<%=  form_with model: @new_group ,url: group_index_path, method: :post do |form|  %>
+  <div class='form-group'>
+    <%= form.text_field :name, class: 'form-control', placeholder: 'Category Name' %>
+    <%= form.text_field :icon , class: 'form-control', placeholder: 'Icon' %>
+    <%= form.submit "Submit", class: 'btn btn-primary' %>
+</div>
+<% end %>
```
Comment (Gambit142) [inline app/views/group/new.html.erb]: - [x] In the categories page, there should be a back button that takes the user to the **_Home Page_** `(group#index)` page. Kindly implement it. 👍

File: app/views/entity/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<%= render partial: 'shared/navbar', locals: { heading: 'CREATE TRANSACTION', back_path: group_entity_index_path } %>
+<%= form_with model:@new_entity, url: group_entity_index_path , method: :post do |form|  %>
+  <div class='form-group'>
+    <%= form.text_field :name, class: 'form-control', placeholder: 'Name' %>
+    <%= form.text_field :amount, class: 'form-control', placeholder: 'Amount' %>
+    <%= form.submit "Submit", class: 'btn btn-primary' %>
+  </div>
+<% end %>
```
Comment (Gambit142) [inline app/views/entity/new.html.erb]: - [x] For the **_add transactions page_**, kindly implement the following features:

1. The form for creating a transaction should have an input field for mapping the transaction being created to a category or categories that already exist. Remember that the relationship between a group and entity is a many-to-many relationship. This means that a group (category) can have many entities (transactions) and an entity (transaction) can have many groups (categories).

2. The back button on this page should take the user to the transactions page `(entity#index page)` and not the categories page `(group#index page)`.

Kindly implement these changes. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @farouqdaif1 👋, 

Great job so far building this ruby on rails application!

## Highlights
- The application has a splash screen ✔️
- The application uses PostgreSQL as a database ✔️ 
- The student used _**Devise**_ for authentication ✔️
- The migration files are set up properly ✔️ 
- Only the necessary routes are added ✔️ 
- Only necessary logic is inside the controllers ✔️
- Application has the necessary view files ✔️  

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- Kindly write **_integration_** tests for the `Group` and `Entity` models 👍 

- _Please check the comments under the review for all other required changes._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/views/group/new.html.erb (hunk 5 above)
Comment (farouqdaif1) [reply app/views/group/new.html.erb]: Thanks for your review task is done

File: app/views/group/index.html.erb (hunk 3 above)
Comment (farouqdaif1) [reply app/views/group/index.html.erb]: Thanks for your review task is done

File: app/views/entity/index.html.erb (hunk 4 above)
Comment (farouqdaif1) [reply app/views/entity/index.html.erb]: Thanks for your review task is done

File: app/views/entity/new.html.erb (hunk 6 above)
Comment (farouqdaif1) [reply app/views/entity/new.html.erb]: Thanks for your review task is done

File: README.md (hunk 1 above)
Comment (farouqdaif1) [reply README.md]: Thanks for your review task is done

File: README.md (hunk 2 above)
Comment (farouqdaif1) [reply README.md]: Thanks for your review task is done

File: README.md (hunk 1 above)
Comment (elisha2kyakpo1) [reply README.md]: - [ ] The deployed app is not working, kindly work on it.

signup

Comment (elisha2kyakpo1) [reply README.md]: > Thanks for your review task is done

- [x] The deployed app is not working, kindly work on it.

signup

File: README.md [hunk 7]
```diff
@@ -2,11 +2,6 @@
 
 "MyBudget" is a budget manager web application that allows users to keep track of their budget with different categories.
 
-
-## Link Demo
-
-[Cick here to get live demo]()
-
 ## Video Demo
 https://www.loom.com/share/122853d2b16a442295145af81cd49a0a
```
Comment (elisha2kyakpo1) [inline README.md]: - [x] The deployed app is not working, kindly work on it.

signup

- [x] Let's put a deployed app live link in the readme file for easy access.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @farouqdaif1, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- You did well to make some changes requested by the previous reviewer ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (farouqdaif1) [reply README.md]: @elisha2kyakpo1 Thanks for your review task is done

File: README.md (hunk 7 above)
Comment (farouqdaif1) [reply README.md]: Thanks for your review task is done

**Review by elyor-doniyorov — APPROVED** [review]
Hi @farouqdaif1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: The live demo link is now working
:heavy_check_mark: The app working as expected

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**